build_projects/dotnet-host-build/PublishTargets.cs

```diff
@@ -164,6 +159,27 @@ public static BuildTargetResult FinalizeBuild(BuildTargetContext c)
             return c.Success();
         }
 
+        [Target(nameof(PrepareTargets.Init),
+        nameof(PublishTargets.InitPublish))]
+        public static BuildTargetResult WriteAzureBlobInfoFile(BuildTargetContext c)
+        {
+            List<string> lines = new List<string>()
+            {
+                "<?xml version=\"1.0\" encoding=\"utf-8\"?>",
+                "<Project ToolsVersion=\"14.0\" DefaultTargets=\"Build\" xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">",
+                "\t<PropertyGroup>",
+                "\t\t<ContainerName>" + Environment.GetEnvironmentVariable("CONTAINER_NAME") + "</ContainerName>",
+                "\t\t<Channel>" + Channel + "</Channel>",
+                "\t\t<BuildNumber>" + SharedFrameworkNugetVersion + "</BuildNumber>",
+                "\t</PropertyGroup>",
+                "</Project>"
+            };
+
+            System.IO.File.WriteAllLines(Dirs.RepoRoot + "\\AzureBlob.props", lines);
```

Comment:
It usually isn't good to write files into the repo during the build.

Why can't these variables be passed on the command-line when we invoke the `syncAzure.proj` file?

Reply:
I could pass ContainerName & Channel from the command line, but that would duplicate what's in BranchInfo.txt. The real issue is BuildNumber, though - this is calculated from CommitCountString, which I wouldn't really have access to from the command line.